app/scripts/controllers/detect-tokens.js

```diff
@@ -91,6 +91,54 @@ export default class DetectTokensController {
     });
   }
 
+  /**
+   * For each token in tokenlist ptovided by the TokenListController, find check selectedAddress balance.
+   */
+  async detectNewTokens() {
+    if (!this.isActive) {
+      return;
+    }
+    if (
```

Comment:
Are binance and mainnet the only two chains that the new token detection api supports? And is there a way to determine supported chains dynamically?

Reply:
I am unsure of that so I changed that to check if the `tokenList` is empty.